proposals/global-grpc-deadline-configuration.md

```diff
@@ -0,0 +1,139 @@
+# Global gRPC Deadline Configuration
+
+**Date Submitted:** 2025-10-17
+
+## Summary
+
+This design proposal introduces a **global `grpcDeadline` configuration** to the **`Client` class** of the Hiero SDKs. The purpose is to align timeout behavior across SDKs, improve consistency, and provide developers with more control over gRPC request handling.
+
+Currently, the Java SDK supports a configurable per-request gRPC deadline via `Client.setGrpcDeadline(Duration)`, while other SDKs (e.g., JavaScript, Go, Rust...) require setting deadlines on individual transactions (e.g., `Transaction.setGrpcDeadline()`). The proposed design adds a **global client-level gRPC deadline property** on the `Client` class, providing a default timeout for all gRPC requests that can be overridden on a per-transaction basis.
+
+This global deadline will also be applied during the **warmup RPC call (initial TCP handshake)** to ensure consistent timeout handling across both initialization and execution flows. The intention is to expose this as a configurable parameter so developers can fine-tune network behavior based on their latency and reliability requirements.
+
+Additionally, this proposal ensures that all SDKs consistently include the **`requestTimeout` property** on the `Client` class. This timeout governs the **overall execution duration** of a `Transaction` or `Query` (including retries, backoff, and node rotation). If not already implemented in a specific SDK, it should be added with a default value of **2 minutes (120,000 ms)**.
+
+This change promotes consistent client behavior across SDKs, simplifies configuration, and enhances usability by allowing global control of both **per-request** and **overall operation** timeouts.
+
+---
+
+## New APIs
+
+### New API #1: `Client.grpcDeadline`
+
+- `number grpcDeadline` : Defines the maximum duration in milliseconds for a gRPC request before timing out.
+  - Default: **10,000 milliseconds (10 seconds)**
+  - Optional property — if not set, the default value applies.
+  - Added as a property of the **`Client` class** across all Hiero SDKs.
+
+#### Methods
+
+- `Client setGrpcDeadline(number grpcDeadlineMs)`  
```

Comment:
What is `number`? `int`/`double`/`time`? This should be clarified all over the doc